cfpb/consumerfinance.gov
# Make translations work for Portal Search Pages

Currently, attempting to use the standard translation molecule on the category views of `PortalSearchPage`s fails because all these pages are associated with the same underlying `pk`. This change extends the `get_translation_links` function such that on `PortalSearchPages` the appropriate category is appended to the generated `href`, if it exists.

## Testing
  - Visit https://www.consumerfinance.gov/consumer-tools/debt-collection/answers/common-issues/
  - Click "Español"
  - Note you're on the top level spanish debt collection page, not the expected `problemas comunes` page
  - Pull this PR
  - Visit localhost:8000/consumer-tools/debt-collection/answers/common-issues/
  - Click "Español"
  - Notice you're on the correct page.
  - Visit localhost:8000/consumer-tools/debt-collection/answers/
  - Click "Español"
  - Notice you're on the right page (that is, this PR didn't break pages that ARE `PortalSearchPage`s but don't have categories
  - Visit http://localhost:8000/consumer-tools/auto-loans/
  - Click "Español"
